gno/r/launchpad_grc20/airdrop_grc20_test.gno

```diff
@@ -0,0 +1,352 @@
+		tokenName     string
+		root          string
+		amountPerAddr uint64
+		start         uint64
+		end           uint64
+	}
+
+	type testNewAirdropExpected struct {
+		panic         bool
+		tokenName     string
+		root          string
+		amountPerAddr uint64
+		start         uint64
+		end           uint64
+	}
+
+	type testNewAirdrop struct {
+		input    testNewAidropInput
+		expected testNewAirdropExpected
+	}
+
+	type testNewAirdropTestTable = map[string]testNewAirdrop
+
+	tests := testNewAirdropTestTable{
+		"Success": {
+			input: testNewAidropInput{
+				tokenName:     "TestNewAirdropMintableToken",
+				root:          "root",
+				amountPerAddr: 100,
+				start:         0,
+				end:           0,
+			},
+			expected: testNewAirdropExpected{
+				panic:         false,
+				tokenName:     "TestNewAirdropMintableToken",
+				root:          "root",
+				amountPerAddr: 100,
+				start:         0,
+				end:           0,
+			},
+		},
+		"Token that does not exist": {
+			input: testNewAidropInput{
+				tokenName:     "ThisTokenDoesNotExist",
+				root:          "root",
+				amountPerAddr: 100,
+				start:         0,
+				end:           0,
+			},
+			expected: testNewAirdropExpected{
+				panic: true,
+			},
+		},
+		"Token that is not mintable": {
+			input: testNewAidropInput{
+				tokenName:     "TestNewAirdropNotMintableToken",
+				root:          "root",
+				amountPerAddr: 100,
+				start:         0,
+				end:           0,
+			},
+			expected: testNewAirdropExpected{
+				panic: true,
+			},
+		},
+		"Airdrop should start in the future or be equal to 0": {
+			input: testNewAidropInput{
+				tokenName:     "TestNewAirdropMintableToken",
+				root:          "root",
+				amountPerAddr: 100,
+				start:         100,
+				end:           0,
+			},
+			expected: testNewAirdropExpected{
+				panic: true,
+			},
+		},
+		"Airdrop should end in the future or be equal to 0": {
+			input: testNewAidropInput{
+				tokenName:     "TestNewAirdropMintableToken",
+				root:          "root",
+				amountPerAddr: 100,
+				start:         0,
+				end:           100,
+			},
+			expected: testNewAirdropExpected{
+				panic: true,
+			},
+		},
+		"Airdrop should start before it ends": {
+			input: testNewAidropInput{
+				tokenName:     "TestNewAirdropMintableToken",
+				root:          "root",
+				amountPerAddr: 100,
+				start:         100,
+				end:           50,
+			},
+			expected: testNewAirdropExpected{
+				panic: true,
+			},
+		},
+	}
+
+	// Create tokens for testing
+	NewToken("TestNewAirdropMintableToken", "TestNewAirdropMintableToken", "noimage", 18, 21_000_000, 23_000_000, true, true)
+	NewToken("TestNewAirdropNotMintableToken", "TestNewAirdropNotMintableToken", "noimage", 18, 21_000_000, 23_000_000, false, true)
+	for testName, test := range tests {
+		t.Run(testName, func(t *testing.T) {
+			if test.expected.panic {
+				defer func() {
+					if r := recover(); r == nil {
+						t.Errorf("Expected panic, got none")
+					}
+				}()
+			}
```

Comment:
we have recently added `p/demo/uassert` & `p/demo/urequire`. these might be useful :)

Reply:
I will look at it, i resolve the comment since PR is already merged and i will look at it to update in a next PR,
thanks for review :pray: